google/apistore.go

```diff
@@ -54,3 +54,29 @@ func (c *APIStore) AddReader(bucket, object, entity string) error {
 	_, err := c.service.ObjectAccessControls.Insert(bucket, object, ac).Do()
 	return err
 }
+
+// AddBucketReader adds a writer of the bucket
+func (c *APIStore) AddBucketReader(bucket, entity string) error {
+	ac := &storage.BucketAccessControl{Entity: entity, Role: "READER"}
+	_, err := c.service.BucketAccessControls.Insert(bucket, ac).Do()
+	return err
+}
+
+// AddBucketWriter adds a writer of the bucket
+func (c *APIStore) AddBucketWriter(bucket, entity string) error {
+	ac := &storage.BucketAccessControl{Entity: entity, Role: "WRITER"}
+	_, err := c.service.BucketAccessControls.Insert(bucket, ac).Do()
+	return err
+}
+
+// SetBucketLifecycle updates a bucket lifecycle policy in days
+func (c *APIStore) SetBucketLifecycle(name string, days int64) error {
+	bucket := &storage.Bucket{Name: name}
+	bucket.Lifecycle = new(storage.BucketLifecycle)
+	action := &storage.BucketLifecycleRuleAction{Type: "Delete"}
+	condition := &storage.BucketLifecycleRuleCondition{Age: days}
+	bucket.Lifecycle.Rule = make([]*storage.BucketLifecycleRule, 1)
+	bucket.Lifecycle.Rule[0] = &storage.BucketLifecycleRule{Action: action, Condition: condition}
+	_, err := c.service.Buckets.Patch(name, bucket).Do()
```

Comment:
Will the bucket lifecycle go in effect once the policy is set? Like, if a bucket is already older than however many `days`, will it be deleted right away?

Reply:
so this is actually an age policy for the objects in the bucket, not the bucket itself. looks like the updated policy can take up to 24 hrs to take affect, and the deletion happens asynchronously, so won't necessarily happen immediately once the condition is satisfied: https://cloud.google.com/storage/docs/lifecycle#behavior

Reply:
nice, I need to add this to other buckets to. I've had a script that added them before, but this will be helpful